mm2src/kdf_walletconnect/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+//! # WalletConnect WebSocket Client
+//!
+//! The WalletConnect client implementation has undergone enhancements to support both WASM and non-WASM targets.
+//! Initially, the implementation was restricted to non-WASM targets,
+//! which limited its utility in web-based environments.
+//!
+//! This module contains the WebSocket client code that has been ported and modified to achieve cross-platform compatibility.
```

Comment:
Also, why did we hard-code/port the entire crate here instead of forking it? Porting the entire source code brings some challanges (such as not being able to track the difference between us and upstream in the future).

Reply:
the library is still in its early stages and not very useful at the moment. Important features like the sign API and session management haven’t been implemented yet, so there might be more changes needed when integrating the sign API in the next iteration. So I think it’s reasonable to proceed with porting it..

cc. @onur-ozkan  @shamardy

Reply:
IMO we can still do the same with forking the repository and maintaining it separately. We can publish it to `crates.io` as well once it's well documented.

Reply:
agreed!